code/processes/dqc.q

```diff
@@ -31,11 +31,14 @@ init:{
   .dqe.loadtimer'[.dqe.configtable];
 
   .dqe.tosavedown:()!();                                                                                        /- store i numbers of rows to be saved down to DB
-  if[.z.p>.eodtime.nextroll:.eodtime.getroll[.z.p];system"t 0";'"next roll is in the past"]                     /- Checking if .eodtime.nextroll is correct
+  if[(`timestamp$.dqe.currentpartition)>=.eodtime.nextroll;.eodtime.getroll[`timestamp$.dqe.currentpartition];.lg.o[`dqc;"Manually update .eodtime.nextroll as it was incorrect"]]
```

Comment:
I think this should be in `.u.end` as the current partition isn't updated until after `init` is called. This should be reverted back to the previous version (check against `.z.p`/`.proc.cp[]`). This will probably fall victim to the same issue as before.